### PR TITLE
[CAFV-449][1.3.z] update CLUSTERCTL.md with updated versions

### PR DESCRIPTION
**Pull request by rliang88**
## Description
Please provide a brief description of the changes proposed in this Pull Request

- add 1.3.z and 1.2.z to table
- updated core capi compatibility to v1.5.4 for 1.3.z and main

## Checklist
- [ ] tested locally
- [ ] updated any relevant dependencies
- [ ] updated any relevant documentation or examples

## API Changes
Are there API changes?
- [ ] Yes
- [ ] No

If yes, please fill in the below

1. Updated conversions?
    - [ ] Yes
    - [ ] No
    - [ ] N/A
2. Updated CRDs?
    - [ ] Yes
    - [ ] No
    - [ ] N/A
3. Updated infrastructure-components.yaml?
    - [ ] Yes
    - [ ] No
    - [ ] N/A
4. Updated `./examples/capi-quickstart.yaml`?
    - [ ] Yes
    - [ ] No
    - [ ] N/A
5. Updated necessary files under `./infrastructure-vcd/v1.0.0/`?
   - [ ] Yes
   - [ ] No
   - [ ] N/A

## Issue
If applicable, please reference the relevant issue

Fixes #

- - -
This change is [Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/614)

### CODE REVIEW

**Review by lzichong — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/614)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @rliang88 and @sahithi)

___
*[`docs/CLUSTERCTL.md` line 22 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/614#-NsysL054ixTXWnnAhSV:-NsysL06C-y_gyliecEg:b-remqoj) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/3cded7ab329a0f40edc95a0d89310f52bddcabd3/docs/CLUSTERCTL.md#L22)):*
> ```Markdown
> 
> 1. Run the below command to initialize the management cluster with the Cluster API and the associated provider for VMware Cloud Director. Ensure the version interop between CAPVCD, Core CAPI, and Clusterctl are correct from the [interop version table](#version-interop)
>    1. For CAPVCD `main`, use `clusterctl init --core cluster-api:v1.4.0 -b kubeadm:v1.4.0 -c kubeadm:v1.4.0 -i vcd:v1.1.0`
> ```

I think the versions should be updated here as well

**Review by lzichong — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/614)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @rliang88 and @sahithi)

___
*[`docs/CLUSTERCTL.md` line 46 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/614#-NsysX7c29zeGjhJw5ax:-NsysX7c29zeGjhJw5ay:br895na) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/3cded7ab329a0f40edc95a0d89310f52bddcabd3/docs/CLUSTERCTL.md#L46)):*
> ```Markdown
> - All of the templates to generate the cluster manifests are located at `templates` directory under the root of the github repository.
> - All the flavors listed support only v1beta1 API versions of CAPVCD and Core CAPI.
> - Currently, we have v1.20.8, v1.21.8, v1.22.9, v1.23.10 as template flavors, and they each have their own etcd/dns versions pre-populated.
> ```

Possibly we can also update this list of versions to the latest kubernetes versions that we have in the templates folder

**Review by lzichong — APPROVED** [review]
:lgtm: after addressing comments

> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/614#-:-Nsysj771cxemyMuA2WJ:b-98fkod)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @rliang88 and @sahithi)

File: docs/CLUSTERCTL.md [hunk 1]
```diff
@@ -1,62 +1,62 @@
...
+| -------------- | ---------------------------- |
+| main           | v1.5.4                       |
+| 1.3.z          | v1.5.4                       |
+| 1.2.z          | v1.4.0                       |
 | 1.1.z          | v1.4.0                       |
 | 1.0.z          | v1.1.3                       |
 
-<a name="clusterctl_set_up"></a>
-## Set up
-Install [clusterctl](https://cluster-api.sigs.k8s.io/user/quick-start.html#install-clusterctl) according to the [version interop table](#version-interop).
```
Comment (sahithi) [inline docs/CLUSTERCTL.md:15]: Please keep this line of clusterctl installation

File: docs/CLUSTERCTL.md [hunk 2]
```diff
@@ -1,62 +1,62 @@
...
-    url: "~/infrastructure-vcd/v1.1.0/infrastructure-components.yaml"
-    type: "InfrastructureProvider"
-```
-
 <a name="init_management_cluster"></a>
+
 ## Initialize Management cluster
+
 1. Run the below command to initialize the management cluster with the Cluster API and the associated provider for VMware Cloud Director. Ensure the version interop between CAPVCD, Core CAPI, and Clusterctl are correct from the [interop version table](#version-interop)
    1. For CAPVCD `main`, use `clusterctl init --core cluster-api:v1.4.0 -b kubeadm:v1.4.0 -c kubeadm:v1.4.0 -i vcd:v1.1.0`
```
Comment (sahithi) [inline docs/CLUSTERCTL.md:22]: vcd:v1.3.0

**Review by sahithi — APPROVED** [review]

**Review by rliang88 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/614)* status: 0 of 1 files reviewed, 4 unresolved discussions (waiting on @lzichong and @sahithi)

___
*[`docs/CLUSTERCTL.md` line 22 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/614#-NsysL054ixTXWnnAhSV:-NtIWJQtCFOJLRPgqBY3:b-896fix) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/3cded7ab329a0f40edc95a0d89310f52bddcabd3/docs/CLUSTERCTL.md#L22)):*
<details><summary><i>Previously, lzichong wrote…</i></summary><blockquote>

I think the versions should be updated here as well
</blockquote></details>

Done.
___
*[`docs/CLUSTERCTL.md` line 46 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/614#-NsysX7c29zeGjhJw5ax:-NtIWJvQ6CQag8JOH1xZ:b-896fix) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/3cded7ab329a0f40edc95a0d89310f52bddcabd3/docs/CLUSTERCTL.md#L46)):*
<details><summary><i>Previously, lzichong wrote…</i></summary><blockquote>

Possibly we can also update this list of versions to the latest kubernetes versions that we have in the templates folder
</blockquote></details>

Done.
___
*[`docs/CLUSTERCTL.md` line 22 at r2](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/614#gh-1526660231:-NtIWHO58efPbHzLEULn:b-896fix) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/332d9c4feadd3739e27a4dc6d63064bb73e3b630/docs/CLUSTERCTL.md#L22)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

vcd:v1.3.0
</blockquote></details>

Done.

**Pull request merged**